contracts/AcceleratingDistributor.sol

```diff
@@ -391,4 +372,31 @@ contract AcceleratingDistributor is ReentrancyGuard, Ownable, Multicall {
             userDeposit.rewardsAccumulatedPerToken = stakingToken.rewardPerTokenStored;
         }
     }
+
+    function _stake(
+        address stakedToken,
+        uint256 amount,
+        address staker
+    ) internal {
+        _updateReward(stakedToken, staker);
+
+        UserDeposit storage userDeposit = stakingTokens[stakedToken].stakingBalances[staker];
+
+        uint256 averageDepositTime = getAverageDepositTimePostDeposit(stakedToken, staker, amount);
```

Comment:
This is existing code but in getAverageDepositTimePostDeposit, I see

uint256 amountWeightedTime = (((amount * 1e18) / (userDeposit.cumulativeBalance + amount)) *
            (getTimeSinceAverageDeposit(stakedToken, account))) / 1e18;

Why not multiple first before dividing by (userDeposit.cumulativeBalance + amount) to minimize rounding errors? Also there are way too many parentheses there.

uint256 amountWeightedTime = (amount * 1e18) * getTimeSinceAverageDeposit(stakedToken, account) / userDeposit.cumulativeBalance + amount) / 1e18;

Reply:
These are good points but I don't love modifying existing already-audited code. @chrismaree wdyt?

Reply:
i dont think we touch this.

Reply:
The rounding error is likely not significant unless a user has staked a significant amount of tokens and then add a very small amount.